full-service/src/json_rpc/txo.rs

```diff
@@ -93,67 +102,64 @@ pub struct Txo {
-    fn from(txo_details: &TxoDetails) -> Txo {
+impl From<&db::models::Txo> for Txo {
+    fn from(txo: &db::models::Txo) -> Txo {
         let mut account_status_map: Map<String, serde_json::Value> = Map::new();
 
-        if let Some(received) = txo_details.received_to_account.clone() {
+        if let Some(received_account_id_hex) = &txo.received_account_id_hex {
+            let txo_status = if txo.is_spent() {
+                TXO_STATUS_SPENT
+            } else if txo.is_pending() {
+                TXO_STATUS_PENDING
+            } else if txo.is_orphaned() {
+                TXO_STATUS_ORPHANED
+            } else {
+                TXO_STATUS_UNSPENT
+            };
+
             account_status_map.insert(
-                received.account_id_hex,
-                json!({"txo_type": received.txo_type, "txo_status": received.txo_status}).into(),
+                received_account_id_hex.to_string(),
+                json!({"txo_type": TXO_TYPE_RECEIVED, "txo_status": txo_status}).into(),
             );
         }
 
-        if let Some(spent) = txo_details.minted_from_account.clone() {
+        if let Some(minted_account_id_hex) = &txo.minted_account_id_hex {
+            let txo_status = if Some(minted_account_id_hex.clone()) != txo.received_account_id_hex {
+                TXO_STATUS_SECRETED
+            } else if txo.is_spent() {
+                TXO_STATUS_SPENT
+            } else if txo.is_pending() {
+                TXO_STATUS_PENDING
+            } else if txo.is_orphaned() {
+                TXO_STATUS_ORPHANED
+            } else {
+                TXO_STATUS_UNSPENT
+            };
+
             account_status_map.insert(
-                spent.account_id_hex,
-                json!({"txo_type": spent.txo_type, "txo_status": spent.txo_status}).into(),
+                minted_account_id_hex.to_string(),
+                json!({"txo_type": TXO_TYPE_MINTED, "txo_status": txo_status}).into(),
             );
         }
 
-        let recipient_address_id = txo_details.txo.recipient_public_address_b58.clone();
-
         Txo {
             object: "txo".to_string(),
-            txo_id_hex: txo_details.txo.txo_id_hex.clone(),
-            value_pmob: (txo_details.txo.value as u64).to_string(),
-            recipient_address_id: if recipient_address_id.is_empty() {
-                None
-            } else {
-                Some(recipient_address_id)
-            },
-            received_block_index: txo_details
-                .txo
-                .received_block_index
-                .map(|x| (x as u64).to_string()),
-            spent_block_index: txo_details
-                .txo
-                .spent_block_index
-                .map(|x| (x as u64).to_string()),
+            txo_id_hex: txo.txo_id_hex.clone(),
+            value_pmob: (txo.value as u64).to_string(),
+            recipient_address_id: None,
+            received_block_index: txo.received_block_index.map(|x| (x as u64).to_string()),
```

Comment:
Why do you need the cast?

Reply:
The database stores these values as i64. This was not actually new code, just moved.